nautiluszim/scraper.py

```diff
@@ -332,19 +336,67 @@ def extract_to_fs(
                     return
                 raise exc
 
-    def test_collection(self):
+    def load_collection(self):
+        """Load the collection.json"""
         with open(self.collection, "r") as fp:
             self.json_collection = [i for i in json.load(fp) if i.get("files", [])]
         nb_items = len(self.json_collection)
         nb_files = sum([len(i.get("files", [])) for i in self.json_collection])
         logger.info(f"Collection loaded. {nb_items} items, {nb_files} files")
 
-        self.test_files()
+    def test_archiveless_collection(self):
+        """Test the collection.json without archive file"""
+        self.load_collection()
+
+        (
+            duplicate_file_names,
+            missing_files,
+            none_url_files,
+            all_file_names,
+        ) = self.test_files()
+
+        if not all_file_names:
+            raise ValueError("Collection is emtpy:\n")
+
+        if none_url_files:
+            raise ValueError(
+                "File(s) referenced in collection are not urls:\n - "
+                + "\n - ".join(none_url_files)
+            )
+
+        self.__test_missing_files(missing_files)
+        self.__test_duplicate_file_names(duplicate_file_names)
 
-    def test_files(self):
+    def test_archive_collection(self):
+        """Test the collection.json with the archive file"""
+        self.load_collection()
         with zipfile.ZipFile(self.archive_path, "r") as zh:
             all_names = zh.namelist()
+        duplicate_file_names, missing_files, _, _ = self.test_files(all_names)
+
+        self.__test_missing_files(missing_files)
+        self.__test_duplicate_file_names(duplicate_file_names)
+
+    def __test_missing_files(self, files):
+        if files:
+            raise ValueError(
+                "File(s) referenced in collection but missing:\n - "
+                + "\n - ".join(files)
+            )
 
+    def __test_duplicate_file_names(self, files):
+        if files:
+            raise ValueError(
+                "Files in collection are duplicate:\n - " + "\n - ".join(files)
+            )
+
+    def test_files(
+        self, all_names: List[str] | None = None
+    ) -> Tuple[Set[str], List[str], List[str], List[str]]:
+        """Tests the file entries and returns:
+        (duplicate_file_names, missing_files, none_url_files, all_file_names)"""
```

Comment:
Some names are not understandable and not described either

Reply:
```py
    def check_collection_filenames(available_filenames: Optional[List[str]] = None) -> tuple()
        """ Check collection filenames for missing URLs or missing member in ZIP file

            all_filenames: list of all target filenames
            missing_filenames: list of entry titles for which a filename is missing
            duplicate_filename: list of target (in ZIM) filenames that are present 2+ times
            local_filenames: list of filenames that are not referencing an URL
```

The list of `non_url_filenames` is used only once so I'd remove it from the function and just loop over `all_filenames` in `test_archiveless_collection` looking for `http`